### DISCUSSION TITLE
Reputation system to asses provider reliability

### DISCUSSION THREAD

**Discussion by vpavlin** [akash-network/community#576] [Ideas]
I just gave a talk about DePIN (and used Akash as an example) at a web2 software architecture event. I got a few questions during the Q&A. They all related to contract, SLA and trust.

To paraphrase: How can I "trust" a random provider to actually deliver the service? Centralized clouds have SLA contracts with users, how is it dealt with in Akash? How can I know a provider will actually provide reliable service?

I went for general answers like "don't trust, verify", trust in decentralized systems is "supported" by economical collateral and then if you don't get reliable service, there is low exit cost to switch to a different provider. I also mentioned the audits Akash is doing.

One thing that came to mind though was that Akash could benefit from a provider reputation system. Few thoughts:

1. The network itself could understand how long a particular lease is running with a provider and could potentially provide some reputation assumptions based on the "churn" - the longer leases the provider has, the better
2. Users could have an option to vote on provider reputation - let's say if I have an active lease  older than N days, I might be eligible to "boost" provider reputation
3. I don't think there should/can be a negative reputation as that could very easily lead to abuse and would be hard to "protocolize".

We ([Waku](waku.org)) are actually looking at decentralized reputation systems for Waku Network, but I am not sure what the sate of that is. It would definitely be easier for Akash since you have the blockchain to record the reputation (which we don't have:) ), but maybe there is a potential for some collaborative exploration.

**Comment by ghost** [comment]
cc @baktun14 @Redm4x @anilmurty

**Comment by vpavlin** [comment]
Cc @jm-clius

**Comment by anilmurty** [comment]
@vpavlin - thanks for the feedback. We've definitely talked about building better provider reputation/ ranking/ scoring system but it's been a combination of priorities and the challenges of ensuring what we build doesn't unfairly penalize some providers. We have a roadmap for Akash provider console (prev. Praetor) that we will add this to https://github.com/akash-network/community/blob/main/product-strategy/praetor-akash-provider-console-strategy.md
